src/components/PokemonTable/PokemonTable.js

```diff
@@ -106,20 +23,8 @@ export default function PokemonTable() {
             </Container>
             <div className={classes.dataGridContainer}>
             <DataGrid className={classes.dataGrid}
-                      onRowClick={
-                          (param ) => {
-                              handleCurrentPokemon(param);
-                              setShowModalWindow(true);
-                          }
-                      }
-                      rows={filteredPokemon.map(pokemon => {
-                          return {
-                              id: pokemon.id,
-                              sprite: pokemon,
-                              name: pokemon.name,
-                              types: pokemon.types.map(pokemonType => pokemonType.type.name)
-                          }
-                      })}
+                      onRowClick={handleRowClick}
```

Comment:
if the name of your props is the same that DataGrid receives you could just use _{...props}_